Fix a typo (#18577)
mdn/content: files/en-us/web/javascript/a_re-introduction_to_javascript/index.md

- HashMaps in Java.
- Associative arrays in PHP.

> The fact that this data structure is so widely used is a testament to its versatility. Since everything (but core types) in JavaScript is an object, any JavaScript program naturally involves a great deal of hash table lookups. It's a good thing they're so fast!

Reviewer: It is not a typo: _bar_ means _except for_ in English.

But as it is a rather British expression, let's go for the clearer _except for_.
```suggestion
The fact that this data structure is so widely used is a testament to its versatility. Since everything (except for core types) in JavaScript is an object, any JavaScript program naturally involves plenty of hash table lookups. It's a good thing they're so fast!
```
